src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -849,6 +1037,13 @@ private void ReportNullabilityMismatchInAssignment(SyntaxNode syntaxNode, object
                         InheritNullableStateOfTrackableStruct(targetType.TypeSymbol, targetSlot, valueSlot, isDefaultValue: IsDefaultValue(value), skipSlot: targetSlot);
                     }
                 }
+
+                // There are some scenarios where we synthesize a bound node for purposes of tracking state, such as in ref/out parameters. In those cases, we
+                // don't want to track the nullability for that parameter as it's entirely fictional and not present in the original bound tree
+                if (!skipAnalyzedNullabilityUpdate)
+                {
+                    SetAnalyzedNullability(value, new VisitResult(valueType, targetType));
```

Comment:
>SetAnalyzedNullability [](start = 20, length = 22)

Why do we need to call `SetAnalyzedNullability` here? I'd expect the callers to have visited the value expression, and thus recorded an entry in the map already.

Reply:
There are lots of cases where they do not.

---
In reply to: [265695540](https://github.com/dotnet/roslyn/pull/34017#discussion_r265695540) [](ancestors = 265695540)

Reply:
How is that possible? This method has a `TypeWithState valueType` argument, which means we must have visited the source expression (and presumably we should have recorded the result then)

---
In reply to: [265697892](https://github.com/dotnet/roslyn/pull/34017#discussion_r265697892) [](ancestors = 265697892,265695540)

Reply:
I'll remove it for now, as I'm not handling tuple deconstruction at the moment and that's what case this was designed to handle, but I believe it'll need to be added back when we add handling for that scenario.

---
In reply to: [265762114](https://github.com/dotnet/roslyn/pull/34017#discussion_r265762114) [](ancestors = 265762114,265697892,265695540)